service/keypairmanager.go

```diff
@@ -33,8 +33,8 @@ const createKeypairTableSQL = `
 		active        boolean default true
 	)
 `
-const listKeypairsSQL = "select id, authority_id, key_id from keypair order by authority_id, key_id"
-const getKeypairSQL = "select id, authority_id, key_id from keypair where id=$1"
+const listKeypairsSQL = "select id, authority_id, key_id, active from keypair order by authority_id, key_id"
+const getKeypairSQL = "select id, authority_id, key_id, active from keypair where id=$1"
 const toggleKeypairSQL = "update keypair set active=$2 where id=$1"
 const upsertKeypairSQL = `
 	WITH upsert AS (
```

Comment:
I know this line is not part this pull request, but it has a bug, so we might fix it while we are at it :smile: 

Actually, the bug is in the line below. You are missing the "WHERE" in the update keypair part of the upsert, so it is always update the existing key, rather than creating a new one.